catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VIsualPlacementActivityTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+import static org.hamcrest.Matchers.equalToIgnoringWhiteSpace;
+
+@RunWith(AndroidJUnit4.class)
+public class VIsualPlacementActivityTest {
+	private int brickPosition;
+	private int xPos = 200;
+	private int yPos = -500;
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(VIsualPlacementActivityTest.class.getSimpleName());
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		brickPosition = 1;
+		PlaceAtBrick brick = new PlaceAtBrick();
+		brick.setCoordinates(xPos, yPos);
+		createProjectAndGetStartScript(PlaceAtBrickTest.class.getSimpleName())
+				.addBrick(brick);
+		baseActivityTestRule.launchActivity();
+	}
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testNoChangeConfirm() {
+
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
+				.perform(click());
+
+		onView(withId(R.id.confirm))
+				.perform(click());
+
+		onView(withId(R.id.brick_place_at_edit_text_x))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(xPos).replace("-", "- ")))));
+		onView(withId(R.id.brick_place_at_edit_text_y))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(yPos).replace("-", "- ")))));
+	}
+
+	@Test
+	public void testNoChangeBackSave() {
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
+				.perform(click());
+
+		Espresso.pressBack();
+
+		onView(withText(R.string.yes))
+				.perform(click());
+
+		onView(withId(R.id.brick_place_at_edit_text_x))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(xPos).replace("-", "- ")))));
+		onView(withId(R.id.brick_place_at_edit_text_y))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(yPos).replace("-", "- ")))));
+	}
+
+	@Test
+	public void testNoChangeDiscard() {
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
+				.perform(click());
+
+		Espresso.pressBack();
+
+		onView(withText(R.string.no))
+				.perform(click());
+
+		onView(withId(R.id.brick_place_at_edit_text_x))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(xPos).replace("-", "- ")))));
+		onView(withId(R.id.brick_place_at_edit_text_y))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(yPos).replace("-", "- ")))));
+	}
```

Comment:
would we need another test which actually makes some changes?

Reply:
From what I gathered, changes would be handled by the VisualPlacementTouchListener and we do have a test which checks if that aspect of the code works properly. 
catroid/src/test/java/org/catrobat/catroid/test/visualplacement/VisualPlacementTouchListenerTest.java
But if you want to I can add a test which does make changes.

Reply:
na, if you say it's tested then it will be fine